pyccel/parser/syntactic.py

```diff
@@ -247,6 +247,56 @@ def _treat_comment_line(self, line, stmt):
         expr.set_fst(stmt)
         return expr
 
+    def _treat_type_annotation(self, stmt, annotation):
+        """
+        Treat an object passed as a type annotation.
+
+        Ensure that an object that was passed as a type annotation can be
+        recognised in the semantic stage by packing it into a SyntacticTypeAnnotation
+        in the correct way. Also check the syntax of any string type
+        annotations.
+
+        Parameters
+        ----------
+        stmt : ast.Ast
+            The ast node about which any errors should be raised.
+
+        annotation : pyccel.ast.basic.PyccelAstNode
+            A visited object which is describing a type annotation.
+
+        Returns
+        -------
+        pyccel.ast.basic.PyccelAstNode
+            The type annotation.
```

Comment:
Maybe it is worth mentioning that the type of the returned object is either `SyntacticTypeAnnotation` or `UnionTypeAnnotation`.

Reply:
Fixed bc2e81a5